### PR TITLE
add java solution to cses 1757

### PR DESCRIPTION
**Pull request by 876pol**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [ ] I have tested my code.
- [ ] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [ ] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/cses-1757.mdx [hunk 1]
```diff
@@ -242,3 +246,77 @@ int main() {
...
+				if (--inDegree[to] == 0) {
+					possibleNodes.add(to);
+				}
+			}
+		}
+
+		for (int i = n - 1; i >= 0; i--) {
+			io.print(ans.get(i) + " ");
+		}
+		io.flush();
```
Comment (SansPapyrus683) [inline solutions/cses-1757.mdx:313]:
```suggestion
```
don't think you need this?

File: solutions/cses-1757.mdx [hunk 2]
```diff
@@ -242,3 +246,77 @@ int main() {
...
+		/*
+		 * Any node with in degree 0 can be used, so we store all possible 
+		 * nodes in a max heap to get the node with the maximum id.
+		*/
+		PriorityQueue<Integer> possibleNodes = new PriorityQueue<>(new Comparator<Integer>() {
+			@Override
+			public int compare(Integer i1, Integer i2) {
+				return -i1.compareTo(i2);
+			}
+		});
```
Comment (SansPapyrus683) [inline solutions/cses-1757.mdx:289]: think you can simplify this w/ a lambda or `Comparator.comparingInt`

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
nice

File: solutions/cses-1757.mdx (hunk 2 above)
Comment (876pol) [reply solutions/cses-1757.mdx:289]: ok just fixed

File: solutions/cses-1757.mdx [hunk 3]
```diff
@@ -242,3 +246,71 @@ int main() {
...
 ```
+
+</CPPSection>
+
+<JavaSection>
+```java
+import java.io.*;
+import java.util.*;
+
+public class Main {
```
Comment (SansPapyrus683) [inline solutions/cses-1757.mdx:257]:
```suggestion
public class CourseScheduleII {
```

**Review by SansPapyrus683 — APPROVED** [review]

File: solutions/cses-1757.mdx [hunk 4]
```diff
@@ -242,3 +246,71 @@ int main() {
 	trav(t,ans) pr(t,' ');
```
Comment (Aadit-Ambadkar) [inline solutions/cses-1757.mdx:246]:
```suggestion
	trav(t, ans) pr(t, ' ');
```

**Review by Aadit-Ambadkar — APPROVED** [review]
beautiful code 
image

**Comment by SansPapyrus683** [comment]
oof forgot to change the class name, oh well

**Pull request merged**